packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -1125,6 +1125,80 @@ public ResponseEntity<JsonNode> makeProjectPublic(
 		}
 	}
 
+	@Operation(summary = "Set a project as a sample project by ID")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "Project has been made a sample project",
+				content = {
+					@Content(
+						mediaType = "application/json",
+						schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = UUID.class)
+					)
+				}
+			),
+			@ApiResponse(
+				responseCode = "403",
+				description = "The current user does not have privileges to modify this project.",
+				content = @Content
+			),
+			@ApiResponse(responseCode = "500", description = "An error occurred verifying permissions", content = @Content)
+		}
+	)
+	@PostMapping("/set-sample/{id}")
+	@Secured(Roles.USER)
+	public ResponseEntity<JsonNode> makeProjectSample(@PathVariable("id") final UUID id) {
+		try {
+			// Only an admin can set a project as a sample project
+			projectService.checkPermissionCanAdministrate(currentUserService.get().getId(), id);
+
+			final Optional<Project> project = projectService.getProject(id);
+			if (project.isEmpty()) {
+				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("projects.not-found"));
+			}
+
+			// Update the project and make it public as well as a sample project
+			project.get().setSampleProject(true).setPublicAsset(true);
+			projectAssetService.togglePublicForAssets(terariumAssetServices, id, true, Schema.Permission.WRITE);
+			projectService.updateProject(project.get());
+
+			/* Permissions */
+
+			// Getting the project
+			final RebacProject rebacProject = new RebacProject(id, reBACService);
+
+			// Delete all previous relationships attached to the project,
+			final List<Contributor> contributors = projectPermissionsService.getContributors(rebacProject);
+			for (final Contributor contributor : contributors) {
+				final RebacUser rebacUser = new RebacUser(contributor.getUserId(), reBACService);
```

Comment:
Need to check if getUserId() returns `null` - that is the signal that the contributor is a group (I guess - been many a passage of time since I wrote that code)

Reply:
I added a test to contributor to check if it's a user or a group and changed wich method is called.
I wish we could simply do this from a RebacObject directly, but that's too much refactoring in one PR.